src/script/lua_api/l_metadata.cpp

```diff
@@ -229,6 +229,56 @@ int MetaDataRef::l_get_keys(lua_State *L)
 	return 1;
 }
 
+// get_bool(self, name)
+int MetaDataRef::l_get_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	IMetadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+    	return luaL_error(L, "The boolean meta value can't be nil");
+	}
+
+	std::string str_;
+	const std::string &str = meta->getString(name, &str_);
+
+	// Explicit check for "true" or "false"
+	if (str == "true") {
+		lua_pushboolean(L, true);
+	} else if (str == "false") {
+		lua_pushboolean(L, false);
+	} else {
+		// Throw error if the value is neither "true" nor "false"
+		return luaL_error(L, "Invalid metadata value. Expected 'true' or 'false'.");
+	}
```

Comment:
This was already flagged on the old PR but if this is meant to match the interface for settings this should really be calling `is_yes`.

Reply:
I don't think this needs to match the setting interface. This data is managed by mods, it isn't directly presented to users. It should be optimized for developer efficiency, and that includes minimizing surprises.

While YAML-isms might make sense if you want something "user-friendly" (though it stands to be argued that the potentially confusing behavior of `is_yes` isn't very user-friendly either), I see little to no value in using them here.

Reply:
tbh have to wonder to what degree this functionality even needs to be part of the engine apart from "trying to match that other API". Mods can implement this with the string or integer functions.

Reply:
I agree with @appgurueu If someone makes a coding error we don’t want to hide it, we should want them to fix it instead. It’ll save their time if something later on changes, anyway. Resolving this conversation, since your request conflicts with @appgurueu ’s which I have already resolved.

Reply:
from "trying to match that other API". Mods can implement this with the string or integer functions.

True, however my argue there is set_bool would in theory use a smaller amount of bytes than set_string. And integer functions have the risk of being misunderstood.